### PR TITLE
feat: category page #1485

### PR DESCRIPTION
**Pull request by lenasokolova**

### CODE REVIEW

**Comment by wlechowicz** [comment]
> Unfortunately, media queries don't work properly and I can't find where the problem is.

I believe they do work properly, you're just not sure about the order. Mobile-first means:
```
export const WrapperGrid = styled.div`
  // any rules defined here apply to mobile and bigger (mobile, tablet, desktop)

  ${device.tablet} {
    // any rules here apply to tablet and bigger (tablet, desktop), overriding mobile rules
  } 

  ${device.desktop} {
    // any rules here apply to desktop and bigger, overriding mobile and tablet rules
  }
`;
```

**Review by rast-mk1 — CHANGES_REQUESTED** [review]
While on `/category/id` page, header elements are unclickable.

**Review by weronika299 — CHANGES_REQUESTED** [review]
I noticed that our category page for popular movies returns  ''oops, looks like there are no movies in this category'' is it ok?

File: app/components/Pages/CategoryPage/CategoryPage.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      <WrapperGrid>
+        {moviesData.map((film, index) => (
+          <MovieCardWrapper key={index}>
+            <MovieCard
+              title={film.title}
+              src={film.images.poster.m}
+              alt={film.title}
+              {...props}
+            ></MovieCard>
+          </MovieCardWrapper>
```
[the remark this answers is not in the document]
Comment (rast-mk1) [reply app/components/Pages/CategoryPage/CategoryPage.js]: Hrefs for movies are `movie/undefined`.

File: pages/category/[id].js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { getListOfMoviesByCategoryId } from 'server/services/movieDb';
+import CategoryPage from 'components/Pages/CategoryPage';
+
+export default function Category({ moviesData }) {
+  return (
+    <div>
```
[the remark this answers is not in the document]
Comment (lenasokolova) [reply pages/category/[id].js:6]: right, it doesn't serve any purpose

File: app/components/Pages/CategoryPage/index.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            id="search"
+            placeholder="Enter the title..."
+            onChange={(event) => setSearchQ(event.target.value)}
+            value={searchQ}
+          />
+        </SearchStyle>
+      </SearchDiv>
+      <WrapperGrid>
+        {filteredList.map((film, id) => (
+          <MovieCardWrapper key={id}>
```
Comment (wlechowicz) [inline app/components/Pages/CategoryPage/index.js:42]: Calling an index an `id` doesn't stop it from being the array's index ;)
```suggestion
        {filteredList.map((film) => (
          <MovieCardWrapper key={film.id}>
```

File: app/components/Pages/CategoryPage/index.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+            <MovieCard
+              href={`/movie/${props.id}`}
+              movieId={film.id}
+              title={film.title}
+              src={film.images.poster.m}
+              alt={film.title}
+              {...props}
```
Comment (wlechowicz) [inline app/components/Pages/CategoryPage/index.js:49]: Are the rest of the props necessary? It's an empty object.

File: app/components/Pages/CategoryPage/index.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+export default function CategoryPage({ moviesData, ...props }) {
+  if (moviesData.length === 0) {
+    return (
+      <ErrorWrapper>
+        <NoMovieError>Oops, looks like there are no movies in this category! :(</NoMovieError>
+      </ErrorWrapper>
+    );
+  }
+
+  const [searchQ, setSearchQ] = useState('');
```
Comment (wlechowicz) [inline app/components/Pages/CategoryPage/index.js:22]: > React Hook "useState" is called conditionally. React Hooks must be called in the exact same order in every component render.eslint[react-hooks/rules-of-hooks](https://reactjs.org/docs/hooks-rules.html)

Seems your ESLint isn't working. Hooks cannot be used conditionally, which means you can't have any conditional return before using hooks. You need to move the ErrorWrapper logic below useState.

File: app/components/Pages/CategoryPage/index.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+            onChange={(event) => setSearchQ(event.target.value)}
+            value={searchQ}
+          />
+        </SearchStyle>
+      </SearchDiv>
+      <WrapperGrid>
+        {filteredList.map((film, id) => (
+          <MovieCardWrapper key={id}>
+            <MovieCard
+              href={`/movie/${props.id}`}
```
Comment (wlechowicz) [inline app/components/Pages/CategoryPage/index.js:44]:
```suggestion
              href={`/movies/${film.id}`}
```

**Review by wlechowicz — CHANGES_REQUESTED** [review]

File: app/components/Pages/CategoryPage/index.js (hunk 3 above)
Comment (lenasokolova) [reply app/components/Pages/CategoryPage/index.js:42]: ok, now I see, I was trying to put `film.id` in `{filteredList.map((film, film.id) => (` and it said there's a mistaken syntax... I thought I have to pass it first as a parameter before putting in a key. My bad.

**Review by weronika299 — APPROVED** [review]

**Review by wlechowicz — APPROVED** [review]
Looks good.
Final cosmetic change would be to set the page container at `min-height: 100vh;` because now a white bottom is showing if the list is short.
obraz

**Review by rast-mk1 — APPROVED** [review]

**Review by marcelina-hasiak — APPROVED** [review]

**Pull request merged**